src/components/Comments/NewCommentForm.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+  ChangeEvent, FC, FormEvent, useCallback, useState,
+} from 'react';
+import cn from 'classnames';
+import { useCreateComment } from '../../hooks/useCreateComment';
+import { Comment } from '../../types/Comment';
+
+type Props = {
+  postId: number;
+};
+
+export const NewCommentForm: FC<Props> = ({ postId }) => {
+  const [name, setName] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [text, setText] = useState<string>('');
+  const [isNameError, setIsNameError] = useState<boolean>(false);
+  const [isEmailError, setIsEmailError] = useState<boolean>(false);
+  const [isTextError, setIsTextError] = useState<boolean>(false);
+  const { mutateAsync, isLoading } = useCreateComment();
+
+  const isAllFilled = name && email && text;
+  const isOneFilled = name || email || text;
+
+  const onReset = useCallback(() => {
+    setName('');
+    setIsNameError(false);
+    setEmail('');
+    setIsEmailError(false);
+    setText('');
+    setIsTextError(false);
+  }, []);
+
+  const onSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    if (!isAllFilled) {
+      return;
+    }
+
+    const newComment: Omit<Comment, 'id'> = {
+      postId,
+      name,
+      email,
+      body: text,
+    };
+
+    try {
+      await mutateAsync(newComment);
+    } catch {
+      return;
+    }
+
+    onReset();
+  };
+
+  const onNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setIsNameError(false);
+    setName(e.target.value);
+  };
+
+  const onEmailChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setIsEmailError(false);
+    setEmail(e.target.value);
+  };
+
+  const onTextChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    setIsTextError(false);
+    setText(e.target.value);
+  };
+
+  const onNameBlur = () => {
+    if (name === '') {
+      setIsNameError(true);
+    }
+  };
+
+  const onEmailBlur = () => {
+    if (email === '') {
+      setIsEmailError(true);
+    }
+  };
+
+  const onTextBlur = () => {
+    if (text === '') {
+      setIsTextError(true);
+    }
+  };
```

Comment:
According [the working version](https://mate-academy.github.io/react_dynamic-list-of-posts/), errors should be shown on submit